sotodlib/io/load_smurf.py

```diff
@@ -611,14 +611,29 @@ def add_file(self, path, session):
                 Path of the file to index
             session : SQLAlchemy session
                 Current, active sqlalchemy session
+            overwrite : bool
+                If true and file exists in the database, update it.
         """
 
         frame_types = {
             ft.type_name: ft for ft in session.query(FrameType).all()
         }
+        
+        ## name has a unique constraint in table
+        db_file = session.query(Files).filter(Files.name==path).one_or_none()
+        if db_file is None:
+            db_file = Files(name=path)
+            session.add(db_file)
+        elif not overwrite:
+            logger.info(f"File {path} found in database, use overwrite=True to update")
+            return
+        else:
+            logger.debug(f"File {path} found in database, updating entry and re-making frames")
+            db_frames = db_file.frames 
+            [session.delete( frame ) for frame in db_frames];
+            session.commit()
```

Comment:
Are we sure we want to commit here? I'd think if the overwrite attempt fails we'd rather leave the db untouched, but could go either way.

Reply:
Yeah, the alternative to committing there felt a lot more complicated and/or more likely to break. 

Instead of deleting, we could go through and update the frames individually, but that would take a much larger re-write of the code or writing an update script that was totally separate from the creation script. The larger re-write was both "work" and I wasn't sure how do to it without adding a whole but of queries to the Frame table that would slow everything down significantly. I didn't like the separate script idea because that's two things we need to update if we change this table.

If we delete and re-create like I'm doing then we have to commit the changes after the delete or the relational mapping gets weird (I'm pretty sure that's true, although I realize now you could convince me I was wrong about that).